owa/app/js/components/manage-autogeneration-option/ManageAutogenerationOption.jsx

```diff
@@ -0,0 +1,455 @@
+import React, { Component } from 'react';
+import { Button, Container, Row, Col, FormGroup, Label, Input, Table,
+    Modal, ModalHeader, ModalBody, ModalFooter, Alert } from 'reactstrap';
+import { MdDelete } from 'react-icons/lib/md';
+import './css/manage-autogeneration.css';
+
+class ManageAutogenerationOption extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      genOptions: [
```

Comment:
Personal preference, try to keep your constructor as light as possible, It looks like you're hardcoding this state, it would be better if you initialize the template of the state in the constructor, then set the state`(this.setState)` in `componentWillMount`. I say this because a lot of times, the first thing you look at in a component is the `constructor` and this one looks really distracting.

Reply:
Yes you are right. I'm currently hardcoding the whole stuff to give a demo overview of what to expect from that component. Definitely the real values will come from the `api endpoints` and I'll switch stuffs around.

Thanks for the observation.